src/test/java/com/ibm/cloud/sdk/core/test/security/IamAuthenticatorTest.java

```diff
@@ -338,6 +338,31 @@ public void testApiErrorBadRequest() throws Throwable {
     PowerMockito.when(Clock.getCurrentTimeInSeconds()).thenReturn((long) 100);
 
     IamAuthenticator authenticator = new IamAuthenticator(API_KEY);
+    authenticator.setURL(url);
+
+    Request.Builder requestBuilder = new Request.Builder().url("https://test.com");
+
+    // Calling authenticate should result in an exception.
+    try {
+      authenticator.authenticate(requestBuilder);
+      fail("Expected authenticate() to result in exception!");
+    } catch (RuntimeException excp) {
```

Comment:
```suggestion
    } catch (ServiceResponseException excp) {
```
Same comment as above.